### PR TITLE
Miscellany

### LINKED ISSUE #35: App boot and dependency loading
Revisit this in the newly structured vue app. Basically we want it to be reasonably efficient, leading to good perceived load time, but also not too complex.

### PR DESCRIPTION
**Pull request by stuartsan**
Closes #35, #36, #28, #37

This does a bunch of things....I'll make some notes inline in the PR.

### CODE REVIEW

File: config/common.js [hunk 1]
```diff
@@ -88,5 +89,11 @@ module.exports = function(env){
         'components': path.resolve(__dirname,'../src/components')
       }
     },
+    plugins: [
```
Comment (stuartsan) [inline config/common.js:92]: Put this back -- the webpack JSON loader was making the file much larger bc of the way it was processing it. Loading it via an xhr.

File: src/assets/styles/index.scss [hunk 2]
```diff
@@ -12,17 +12,28 @@ body {
 }
 
 .btn {
-  background: $light-blue;
+  background: $blue;
```
Comment (stuartsan) [inline src/assets/styles/index.scss:15]: Lots of random styling things to move closer to the designs

Comment (stuartsan) [reply src/assets/styles/index.scss:15]: Also -- the styles are very mobile first

File: src/components/EventCard.vue [hunk 3]
```diff
@@ -0,0 +1,45 @@
+<template>
```
Comment (stuartsan) [inline src/components/EventCard.vue:1]: Some new vue components

File: src/components/EventDateFilters.vue [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    },
+    endDate() {
+      this.filters.endDate;
+    }
+  },
+  methods: {
+    updateSelectedStartDate(event) {
+      const newDate = event.target.value;
+      // FIXME ha ha
+      if (/\d{4}-\d{2}-\d{2}/.test(newDate)) {
```
Comment (stuartsan) [inline src/components/EventDateFilters.vue:37]: Need to plug in some kind of date picker here, this is just a real sketchy date input thing for the moment

File: src/templates/EventList.html [hunk 5]
```diff
@@ -1,5 +1,11 @@
-<div id="event-list">
-  <div v-for="event in events" class="event-card">
-    {{event.title}}
+<div id="event-list" class="event-list" v-show="view === 'list'">
+  {{ filteredEvents.length ? '' : '' }}
```
Comment (stuartsan) [inline src/templates/EventList.html:2]: The stuff I've done here and in the event map html file is bananas -- I noticed that the components were not being updated when the store's values being piped into them changed, unless those values were actually referenced in the template. I just don't know Vue very well, there must be a real way to achieve this, anybody know?

File: src/components/EventMap.js [hunk 6]
```diff
@@ -1,114 +1,100 @@
 import Vue from 'vue';
-import { pollForNewEvents, loadEvents } from 'src/util/events-poll';
+import { getFilteredEvents } from 'src/util/events';
+
+// Attaches L (leaflet.js) to window.
+import 'mapbox.js';
```
Comment (stuartsan) [inline src/components/EventMap.js:5]: I think the code splitting is cool and could be promising but overall I don't think it was giving us much in this case, loading mapbox asynchronously, cause we really want mapbox as soon as possible to be able to render a map while the larger data payloads are arriving asynchronously.

File: config/common.js (hunk 1 above)
Comment (ryanhobsonsmith) [reply config/common.js:92]: Makes sense to me

File: src/templates/EventList.html (hunk 5 above)
Comment (ryanhobsonsmith) [reply src/templates/EventList.html:2]: Hmm, yeah that doesn't sound right to me. I'll try to take a look later today but have to be in a meeting now.

File: src/components/EventMap.js (hunk 6 above)
Comment (ryanhobsonsmith) [reply src/components/EventMap.js:5]: Yup, I agree.

**Review by ryanhobsonsmith — APPROVED** [review]
Generally looks good

**Pull request merged**